### DISCUSSION TITLE
Above head text chat

### DISCUSSION THREAD

**Discussion by Kael2244** [ParadiseSS13/Paradise#15922] [Code Help]
I would like to know where in the code is exactly that feature we have in Paradise , mostly to learn how it works since I don't understand it.

**Comment by Sirryan2002** [comment]
The feature is called "runechat".  From what I know both of these files contain most of the runechat code, however other people might know more than I do so feel free to add to this.
[code/controllers/subsystem/runechat.dm](https://github.com/ParadiseSS13/Paradise/blob/f0841b36294baccc2c5ac18678bd898defb8583d/code/controllers/subsystem/runechat.dm)
[code/datums/chatmessage.dm](https://github.com/ParadiseSS13/Paradise/blob/f0841b36294baccc2c5ac18678bd898defb8583d/code/datums/chatmessage.dm)

**Comment by AffectedArc07** [comment]
Original PR to add it
https://github.com/ParadiseSS13/Paradise/pull/14141
